src/utilities/network/lookupApi.js

```diff
@@ -22,88 +23,22 @@ const SEPARATORS = {
   POLICY_NUMBER: '-',
 };
 
-export const createPatientRecord = response => {
-  const { 
-    id,
-    name,
-    code,
-    phoneNumber,
-    billAddress1,
-    billAddress2,
-    billAddress3,
-    billAddress4,
-    billPostalZipCode,
-    emailAddress,
-    type,
-    isCustomer,
-    isSupplier,
-    isManufacturer,
-    supplyingStoreId,
-    thisStoresPatient,
-    isPatient,
-    firstName,
-    lastName,
-    dateOfBirth,
-  } = response;
-  const billingAddress = getOrCreateAddress(
-    UIDatabase,
-    billAddress1,
-    billAddress2,
-    billAddress3,
-    billAddress4,
-    billPostalZipCode
-  );
-  const patient = {
-    id,
-    name,
-    code,
-    phoneNumber,
-    billingAddress,
-    emailAddress,
-    type,
-    isCustomer,
-    isSupplier,
-    isManufacturer,
-    supplyingStoreId,
-    thisStoresPatient,
-    isPatient,
-    firstName,
-    lastName,
-    dateOfBirth,
-    isVisible: true,
-  };
-  UIDatabase.write(() => UIDatabase.update('Name', patient));
+export const createPatientRecord = patient => {
+  UIDatabase.write(() => createRecord(UIDatabase, 'Patient', patient));
+  patient?.policies?.forEach(createPolicyRecord);
 };
 
-export const createPrescriberRecord = response => {
-  const {
-    id,
-    firstName,
-    lastName,
-    registrationCode,
-    address1,
-    address2,
-    phoneNumber,
-    mobileNumber,
-    emailAddress,
-    fromThisStore,
-  } = response;
-  const address = getOrCreateAddress(UIDatabase, address1, address2);
-  const prescriber = {
-    id,
-    firstName,
-    lastName,
-    registrationCode,
-    address,
-    isVisible: true,
-    isActive: true,
-    phoneNumber,
-    mobileNumber,
-    emailAddress,
-    fromThisStore,
-  };
-  UIDatabase.write(() => UIDatabase.update('Prescriber', prescriber));
-}
+export const createPrescriberRecord = prescriber =>
+  UIDatabase.write(() => createRecord(UIDatabase, 'Prescriber', prescriber));
+
+export const createPolicyRecord = policy => {
+  const { nameId, enteredById, insuranceProviderId } = policy;
+  const enteredBy = UIDatabase.getOrCreate('User', enteredById);
+  const patient = UIDatabase.getOrCreate('Name', nameId);
+  const insuranceProvider = UIDatabase.getOrCreate('InsuranceProvider', insuranceProviderId);
+  const policyRecord = { ...policy, enteredBy, patient, insuranceProvider };
+  UIDatabase.write(() => createRecord(UIDatabase, 'InsurancePolicy', policyRecord));
```

Comment:
I dont know exactly where this is being called from, but I generally try to keep writes to the most top of a call stack as I can as you can't start a new write from within one, so it can avoid making changes in the future and subtle/annoying bugs.  Although not too important, creating a new transaction is quite a high proportion of the time spent doing a write, best to try and do as many actions within the same transaction as pos imo

Reply:
Good idea, about to do the thunk work so thinking best solution might be to wrap the policy logic in the same write transaction as the patient.